src/InverseKinematics.cpp

```diff
@@ -0,0 +1,154 @@
+#include "InverseKinematics.h"
+
+
+InverseKinematics::InverseKinematics()
+:updated(false)
+,initialized(false)
+,alreadyOptimized(false)
+{
+    solverPointer = new InverseKinematicsIPOPT();
+    loader = IpoptApplicationFactory();
+    loader->Options()->SetStringValue("hessian_approximation", "limited-memory");
+}
+
+InverseKinematics::~InverseKinematics()
+{
+}
+
+bool InverseKinematics::prepareProblem(const std::string& filename, const std::vector< std::string >& consideredJoints, const iDynTree::Vector3& gains, const iDynTree::Vector3& desiredPosition, const iDynTree::Vector4& desiredQuaternion, const iDynTree::VectorDynSize& desiredJoints, const std::string& parentFrame, const std::string& endEffectorFrame)
+{
+    bool success;
+    success = solverPointer->loadFromFile(filename, consideredJoints);
```

Comment:
The error is the other class :wink:, which has much more information about models and stuff.